core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -416,6 +418,57 @@
    void putForExternalRead(K key, V value, Metadata metadata);
 
    /**
+    * An overloaded form of {@link #compute(K, BiFunction)}, which takes in an
+    * instance of {@link Metadata} which can be used to provide metadata
+    * information for the entry being stored, such as lifespan, version
+    * of value...etc. The {@link Metadata} is only stored if the call is
```

Comment:
There are cases where the method computeIfPresent might not be successful but this is more the case in computeIfPresent and computeIfAbsent, as no metadata is updated. I'm removing the line